decidim/decidim
# Redirect non-admin users to core's root_path

#### :tophat: What? Why?

While reviewing #10117, @ahukkanen [found out this issue](https://github.com/decidim/decidim/pull/10117#discussion_r1219019026):

> We need one more condition here because now regular users would be redirected to a 404 page in case they try to access an admin view.

I preferred to handle this in another PR just in case there are lots of specs to change.

#### :pushpin: Related Issues
 
- Related to #10117

#### Testing

~1. Sign in as Assembly's Admin ("assembly_1_admin@example.org")~
~2. Go to an admin page that you're not authorized, such as http://localhost:3000/admin/participatory_processes~
~3. See that you're redirected to the core's root_path ("/") instead of the admin's root_path ("/admin")~
(This was not actually a bug, [read my comment with the full explanation](https://github.com/decidim/decidim/pull/10968#issuecomment-1647408497))

Without the patch:

1. Login as `user@example.org`
2. Type in the URL `/admin/templates/questionnaire_templates`
3. Get redirected to the admin TOS page which gives you a 404
4. Apply the patch
5. See that you're redirected to the /admin page

:hearts: Thank you!
